### PR TITLE
Adding additional claims to Identity

### PR DESCRIPTION
**Pull request by damienbod**
fixes https://github.com/dotnet/AspNetCore.Docs/issues/9244

@scottaddie @Rick-Anderson 

[Internal review page](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/add-user-data?view=aspnetcore-1.0&branch=pr-en-us-17394&tabs=visual-studio#add-claims-to-identity-using-iuserclaimsprincipalfactory)

### CODE REVIEW

File: aspnetcore/security/authentication/add-user-data.md [hunk 1]
```diff
@@ -222,3 +222,81 @@ Test the app:
...
+		identity.AddClaims(claims);
+		return principal;
+	}
+}
+
+```
+
+The additional claim can then be used in the application. In a Razor Page, the `IAuthorizationService` instance can be used to access the claim value.
+
+```csharp
```
Comment (scottaddie) [inline aspnetcore/security/authentication/add-user-data.md]:
```suggestion
```cshtml
```

File: aspnetcore/security/authentication/add-user-data.md [hunk 2]
```diff
@@ -222,3 +222,81 @@ Test the app:
 * Register a new user.
 * View the custom user data on the `/Identity/Account/Manage` page.
 * Download and view the users personal data from the `/Identity/Account/Manage/PersonalData` page.
+
+# Add claims to Identity using IUserClaimsPrincipalFactory<ApplicationUser>
+
+Additional claims can be added to ASP.NET Core Identity by using the `IUserClaimsPrincipalFactory<T>` interface. 
+This class can be added to the application in the `ConfigureServices` method. Add the custom implementation of the class as follows:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/add-user-data.md]:
```suggestion
Additional claims can be added to ASP.NET Core Identity by using the `IUserClaimsPrincipalFactory<T>` interface. This class can be added to the app in the `Startup.ConfigureServices` method. Add the custom implementation of the class as follows:
```

File: aspnetcore/security/authentication/add-user-data.md [hunk 3]
```diff
@@ -222,3 +222,81 @@ Test the app:
...
+
+The demo code uses the `ApplicationUser` class. This class adds an `IsAdmin` property which is used to add the additional claim.
+
+```csharp
+public class ApplicationUser : IdentityUser
+{
+	public bool IsAdmin { get; set; }
+}
+```
+The `AdditionalUserClaimsPrincipalFactory` implements the `UserClaimsPrincipalFactory` interface. A new role claim is added to the `ClaimsPrincipal`.
```
Comment (scottaddie) [inline aspnetcore/security/authentication/add-user-data.md]:
```suggestion

The `AdditionalUserClaimsPrincipalFactory` implements the `UserClaimsPrincipalFactory` interface. A new role claim is added to the `ClaimsPrincipal`.
```

File: aspnetcore/security/authentication/add-user-data.md [hunk 4]
```diff
@@ -222,3 +222,81 @@ Test the app:
...
+		}
+
+		identity.AddClaims(claims);
+		return principal;
+	}
+}
+
+```
+
+The additional claim can then be used in the application. In a Razor Page, the `IAuthorizationService` instance can be used to access the claim value.
```
Comment (scottaddie) [inline aspnetcore/security/authentication/add-user-data.md]:
```suggestion
The additional claim can then be used in the app. In a Razor Page, the `IAuthorizationService` instance can be used to access the claim value.
```

Comment (damienbod) [reply aspnetcore/security/authentication/add-user-data.md]: fixed

File: aspnetcore/security/authentication/add-user-data.md (hunk 3 above)
Comment (damienbod) [reply aspnetcore/security/authentication/add-user-data.md]: fixed

File: aspnetcore/security/authentication/add-user-data.md (hunk 2 above)
Comment (damienbod) [reply aspnetcore/security/authentication/add-user-data.md]: fixed

File: aspnetcore/security/authentication/add-user-data.md (hunk 1 above)
Comment (damienbod) [reply aspnetcore/security/authentication/add-user-data.md]: fixed

**Comment by scottaddie** [comment]
Paging @blowdart for a review.

**Comment by scottaddie** [comment]
Thank you once again for helping us out, @damienbod! Much appreciated.

**Comment by sbsw** [comment]
Curious minds want to know: Is there an easy way to get from this GitHub page to the Microsoft website page that was updated by this issue closure?

**Comment by Rick-Anderson** [comment]
@sbsw  see #9244, the issue

**Pull request merged**